src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  // died() {
+  //   if (this.health <= 0) {
+  //     delete Animal.alive.this;
+  //   }
+  // }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.name = name;
+  }
+
+  bite(name) {
+    if (name.hidden === false) {
+      name.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter((element) => element.health !== 0);
```

Comment:
Using `filter` to remove dead animals from the `Animal.alive` array after each bite is inefficient. Consider moving this logic to a separate method and calling it when an animal's health drops to 0 or less to avoid unnecessary iterations over the array.